test/framework/http.go

```diff
@@ -88,13 +103,73 @@ func ExecuteHTTPRequestWithUnmarshalledResponse(namespace, httpMethod, uri, path
 
 // IsHTTPRequestSuccessful makes and checks whether an http request is successful
 func IsHTTPRequestSuccessful(namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
-	response, err := ExecuteHTTPRequest(namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+	return IsHTTPRequestSuccessfulC(&http.Client{}, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+}
+
+// IsHTTPRequestSuccessful makes and checks whether an http request is successful using a given client
+func IsHTTPRequestSuccessfulC(client *http.Client, namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
+	response, err := ExecuteHTTPRequestC(client, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
 	if err != nil {
 		return false, err
 	}
+	io.Copy(ioutil.Discard, response.Body)  // Just read the response to be able to close the connection properly
+	defer response.Body.Close()
 	return checkHTTPResponseSuccessful(namespace, response), nil
 }
 
+// ExecuteHTTPRequestsInThreads executes given number of requests using given number of threads (Go routines).
+// Returns []HTTPRequestResult with the outcome of each thread (HTTPRequestResultSuccess or HTTPRequestResultError).
+// Useful for performance testing.
+func ExecuteHTTPRequestsInThreads(namespace, httpMethod string, requestCount, threadCount int, routeURI, path, bodyFormat, bodyContent string) []HTTPRequestResult {
+	requestPerThread := requestCount / threadCount
```

Comment:
Yes, considered it, but I would assume that nobody sane would specify 13 threads running 1000 requests. Anyway, in that case I would have to consider rounding it which means the metrics won't be accurate anyway. So I though we can use common sense here :)
But if you want, I can put there a panic in case the remainder is non-zero.

Reply:
ahah indeed.
a small panic or return error would be good so we know why it fails.
if not, someone may spend some time in the future just to figure that out (worst-case scenario)XD

Reply:
OK, done.